internal/internal_worker_base.go

```diff
@@ -401,3 +410,38 @@ func (bw *baseWorker) Stop() {
 	}
 	return
 }
+
+func (bw *baseWorker) EmitHardwareUsage() {
+	defer bw.shutdownWG.Done()
+
+	emitOnce.Do(func() {
+		go func() {
+			ticker := time.NewTicker(hardwareEmitInterval)
```

Comment:
This needs a panic recover https://go.dev/blog/defer-panic-and-recover

Reply:
Added recover